docs/whats-new/preview-4.md

```diff
@@ -0,0 +1,647 @@
+          "scheme": "https",
+          "protocol": "tcp",
+          "transport": "http"
+          }
+      }
+    }
+  }
+}
+```
+
+As you can see, the manifest captures both the parameters and the usage of the bicep output in the environment variables of the project.
+
+## Emulators
+
+In .NET Aspire preview 3, we introduced the ability to run emulators for various services. We've changed the APIs to match with the idioms described earlier in this document:
+
+**Azurite: Azure storage emulator**
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var blobs = builder.AddAzureStorage("storage")
+                   .RunAsEmulator()
+                   .AddBlobs("blobs");
+
+builder.AddProject<Projects.Api>("api")
+       .WithReference(blobs);
+
+builder.Build().Run();
+```
+
+Each `RunAsEmulator` method has a callback that enables customization of the emulator container resource. For example, you can change the image, tag, or add additional volumes.
+
+**Azure storage emulator (Azurite)**
+
+```csharp
+var storage = builder.AddAzureStorage("storage").RunAsEmulator(container =>
+{
+    container.UsePersistence();
+});
+```
+
+**CosmosDB emulator**
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var db = builder.AddAzureCosmosDB("cosmos")
+                .RunAsEmulator()
+                .AddDatabase("db");
+
+builder.AddProject<Projects.Api>("api")
+       .WithReference(db);
+
+builder.Build().Run();
+```
+
+## The deployment manifest
+
+The aspire manifest has undergone a significant overhaul to support the new primitives and introduced in preview 4. The new manifest focuses on a few key primitives to enable a wide range of scenarios:
+
+| Resource type   | Description                                                   |
+|-----------------|---------------------------------------------------------------|
+| `project.v0`    | .NET project files                                            |
+| `container.v0`  | Container images                                              |
+| `dockerfile.v0` | Docker files                                                  |
+| `parameter.v0`  | External Parameters                                           |
+| `value.v0`      | References to other resources (or a combination of resources) |
+
+### Azure specific resources
+
+| Resource type    | Description           |
+|------------------|-----------------------|
+| `azure.bicep.v0` | Azure Bicep templates |
+
+Tool authors can support this very small set of resource types to model lots of different apps!
+
+We've deprecated the abstract resource types that were supported in previous versions. Deployment tools may still support those resource types, but they are no longer part of the core manifest schema.
+
+### Visual Studio Publish to Azure
+
+This release we've enabled a Visual Studio publish experience for Aspire applications.
+
+TODO: Deployment to Azure via Visual Studio support (@bradygaster @timheuer)
```

Comment:
@timheuer @bradygaster Do we have any content for this? We have another week for content so it would be great if we could write a blurb about this.